server/services/caseListService.js

```diff
@@ -30,19 +35,18 @@ module.exports = function createCaseListService(nomisClientBuilder, roService, l
   /**
    * Assume username is assigned to an RO.  Look up this user in the local db (staff_ids table) and take the staff code.
    * Alternatively if a unique staff code cannot be found in this db, ask Delius for a staff code.
-   *
-   * The code below is an ( ugly bodge / elegant solution ) that breaks the type contract on Result.orElse():
-   * staffCodeFromDB is a Result<string,string>, but getStaffCodeFromDelius() returns a Promise<Result<string,string>>
-   * This means that the .orElse() will return either a Result<string,string> or a Promise<Result<string,string>>
-   * Fortunately the 'await' prefix flattens nested Promises so that it all 'just works'.
-   *
    */
   const getROCaseList = (username, token) => async () => {
     const staffCodeFromDb = await getStaffCodeFromDb(username)
-    const staffCode = await staffCodeFromDb.orElse(() => getStaffCodeFromDelius(username))
+    const staffCode = await staffCodeFromDb.orElseTryAsync(() => getStaffCodeFromDelius(username))
     return staffCode.map(getOffendersForStaffCode(token)).match(R.identity, message => ({ hdcEligible: [], message }))
```

Comment:
so now intelliJ sees staffCode as type `Result<String, String>`.
Still need to add type info for the curried function `getOffendersForStaffCode`. 

This currently only compiles as `R.identity` moves everything over to `any`